Classes/Hooks/DataHandler.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+ * @license GPLv3, http://www.gnu.org/copyleft/gpl.html
+ * @copyright Gilbertsoft (gilbertsoft.org), 2017
+ * @copyright Aimeos (aimeos.org), 2017-
+ * @package TYPO3
+ */
+
+namespace Aimeos\Aimeos\Hooks;
+
+
+use TYPO3\CMS\Backend\Utility\BackendUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+
+/**
+ * Hooks for DataHandler
+ *
+ * @package TYPO3
+ */
+class DataHandler
+{
+
+	/**
+	 * Templates, please insert in alphabetical order if used!
+	 *
+	 * public function processCmdmap_preProcess( $command, $table, $id, $value, \TYPO3\CMS\Core\DataHandling\DataHandler &$pObj ) {}
+	 * public function processCmdmap_postProcess( $command, $table, $id, $value, \TYPO3\CMS\Core\DataHandling\DataHandler &$pObj ) {}
+	 * public function processCmdmap_deleteAction( $table, $id, $recordToDelete, $recordWasDeleted=NULL, \TYPO3\CMS\Core\DataHandling\DataHandler &$pObj ) {}
+	 * public function processDatamap_preProcessFieldArray( array &$fieldArray, $table, $id, \TYPO3\CMS\Core\DataHandling\DataHandler &$pObj ) {}
+	 * public function processDatamap_afterAllOperations( \TYPO3\CMS\Core\DataHandling\DataHandler &$pObj ) {}
+	 * public function processDatamap_afterDatabaseOperations( $status, $table, $id, array $fieldArray, \TYPO3\CMS\Core\DataHandling\DataHandler &$pObj ) {}
+	 *
+	 */
+
+	/**
+	* Checks if the fields defined in $deprecatedFields are set in the data-array of pi_flexform and unsets it.
+	*
+	* Structure of the deprecatedFields array:
+	*
+	* [ 'list_type' => ['field1', 'field2'] ];
+	*
+	* @param string $status
+	* @param string $table
+	* @param string $id
+	* @param array $fieldArray
+	* @param \TYPO3\CMS\Core\DataHandling\DataHandler $pObj
+	*
+	* @return void
+	*/
+	public function processDatamap_postProcessFieldArray( $status, $table, $id, array &$fieldArray, \TYPO3\CMS\Core\DataHandling\DataHandler &$pObj )
+	{
+		// Only listen to flexform data
+		if( $table === 'tt_content' && $status === 'update' && isset( $fieldArray['pi_flexform'] ) )
+		{
+			// Get list type
+			$listType = isset( $fieldArray['list_type'] ) ? $fieldArray['list_type'] : '';
+
+			if( empty( $listType ) )
+			{
+				$record = BackendUtility::getRecord( $table, $id, 'list_type' );
+				$listType = $record['list_type'];
+			}
+
+
+			// Only listen to own plugin data
+			if( !empty( $listType ) && compare( $listType, 'aimeos_', 1) == 0 )
+			{
+				// Define deprecated fields which should be ignored on saving
+				$deprecatedFields = array(

```

Comment:
Currentyl not used, but a simple way to cleanup some settings. Perhaps useful in the future